linera-core/src/client/mod.rs

```diff
@@ -675,21 +677,23 @@ where
     /// local chain.
     #[tracing::instrument(level = "trace")]
     async fn pending_message_bundles(&self) -> Result<Vec<IncomingBundle>, ChainClientError> {
-        if self.next_block_height() != BlockHeight::ZERO && self.options.message_policy.is_ignore()
-        {
-            return Ok(Vec::new()); // OpenChain is already received, other are ignored.
-        }
         let query = ChainInfoQuery::new(self.chain_id).with_pending_message_bundles();
         let info = self
             .client
             .local_node
             .handle_chain_info_query(query)
             .await?
             .info;
-        ensure!(
-            info.next_block_height == self.next_block_height(),
-            ChainClientError::WalletSynchronizationError
-        );
+        {
+            let state = self.state();
+            ensure!(
+                state.has_other_owners(&info.manager.ownership) || info.tip() == state.tip(),
```

Comment:
Question: I'm not sure I understand why this new condition was added :sweat_smile:

Reply:
Because if the chain has other owners it could have been extended without us knowing about it. So our chain state we read from our wallet not matching the actual state is not an error.

Reply:
Seems like but we may just want a warning. The chain could have other owners now unbeknownst to us.

Reply:
I'd imagine we need to learn (sync) the chain owners before we process its blocks?

Reply:
That's true, this would still fail if we deleted our storage.

(At least if we don't, it can't: If at the highest block known to us we are still the only owners, then ownership can't have changed.)